power_grid_model_c/power_grid_model/include/power_grid_model/main_model.hpp

```diff
@@ -198,6 +198,21 @@ class MainModelImpl<ExtraRetrievableTypes<ExtraRetrievableType...>, ComponentLis
                 update[entry.index](*this, found->second, pos, sequence_idx_map[entry.index]);
             }
         }
+
+        // sequence_idx_map accumulation logic
```

Comment:
i don't know if you want to do that in a follow-up PR, but only the elements that either affect `UpdateChange::topo` or `UpdateChange::param` need to be added. There are types of update that affect neither. E.g.:

* sensor updates do not affect Ybus and therefore do not require Ybus to be updated (right-hand side of equation)
* status updates do not affect parameters, only topology. if an update keeps the same statuses (e.g. from 1 to 1), then the topology does not change and therefore there's no need to rebuild everything
* a load or generator only affects matrix entries, not ybus
* a shunt update **does** affect ybus if it changes any of the shunt values

You can find how and where this logic is used by doing a global search for `UpdateChange`

Reply:
Let's definitely put it in the next PR -- there will always be unknowns (to me) popping up

Reply:
true but i'll have to think about whether this will have any significant performance impact. the key here is when it does the update param. i'll dig some into this later today

Reply:
as discussed, I changed a significant portion of the implementation of the main model to account for these changes, as well as the consecutive sym/asym calculations edge case.

Do you think you can create a main model unit test case for that? you only need to verify that at any stage, the calculation is the same as if you would choose a different order in this scenario:

- initialize
- sym pf
- asym pf
- perm update
- perm update
- asym pf
- sym pf

i think you then have all edge cases, because in the naive implementation, the last sym pf calculation does not change parameters, but in the new implementation, it does

As for implementation details: I now bypass querying new calc params against ybus calc params. next optimization is to move away from calculation all calc params in the main model and only reuse the one from ybus

Reply:
Resolved in the new fix commit